searchableSelect.js

```diff
@@ -0,0 +1,203 @@
+            // event handler will take care of the rest.
+            obj.input.blur();
+
+            // Select can happen instantly when pressing enter.
+            if (obj.selectTimeout) {
+                clearTimeout(obj.selectTimeout);
+                obj.selectTimeout = null;
+                obj.applySelection();
+            }
+
+        } else if (event.key === 'ArrowUp' || event.key === 'ArrowDown') {
+            // Don't let the cursor be moved by the key press
+            event.preventDefault();
+
+            if (selected) {
+                let newSelect = selected;
+    
+                if (event.key === 'ArrowUp') {
+                    // Get the next list item up that isn't hidden (if there is one)
+                    do {
+                        newSelect = newSelect.previousSibling;
+                    } while (newSelect && newSelect.classList.contains('hidden'));
+                } else {
+                    // Get the next list item down that isn't hidden (if there is one)
+                    do {
+                        newSelect = newSelect.nextSibling;
+                    } while (newSelect && newSelect.classList.contains('hidden'));
+                }
+    
+                // If there is a next list item up/down, make it the new selected item
+                if (newSelect) {
+                    selected.classList.remove('selected');
+                    newSelect.classList.add('selected');
+                    newSelect.scrollIntoView({block: 'nearest'});
+                }
+            }
+        }
+    });
+}
+
+SearchableSelect.prototype.search = function(query) {
+    let search;
+
+    if (typeof(query) === 'string') {
+        this.input.value = query;
+        search = query.toLowerCase();
+    } else {
+        search = this.input.value.toLowerCase();
+    }
+
+    // Hide the list items that do not have the search query
+    for (let child of this.list.children) {
+        if (child.innerText.toLowerCase().includes(search)) {
+            child.classList.remove('hidden');
+        } else {
+            child.classList.add('hidden');
+        }
+    }
+
+    // Un-select the currently selected list item
+    for (let i = 0; i < this.list.children.length; i++) {
+        if (this.list.children[i].classList.contains('selected')) {
+            this.list.children[i].classList.remove('selected');
+            break;
+        }
+    }
+
+    // Select the first list item that is not hidden (or none if there are none)
+    for (let i = 0; i < this.list.children.length; i++) {
+        if (!this.list.children[i].classList.contains('hidden')) {
+            this.list.children[i].classList.add('selected');
+            this.list.children[i].scrollIntoView({ block: 'nearest' });
+            break;
+        }
+    }
+}
+
+SearchableSelect.prototype.getSelectedObj = function() {
+    let selected = null;
+
+    for (let i = 0; i < this.list.children.length; i++) {
```

Comment:
I think we should think about standardizing on either ES6 or ES5 syntax, it feels like you're mixing it a lot